### PR TITLE
fix(launch profiling): debug logs and cleanup

### PR DESCRIPTION
**Pull request by armcknight**
Follows https://github.com/getsentry/sentry-cocoa/pull/3634 and followed by https://github.com/getsentry/sentry-cocoa/pull/3636

Remove some global variables that are no longer needed now that we use a dedicated transaction for app launch. 

Remove some old API from the backup config file strategy we no longer use. It was causing some build failures in tests.

Also some debug logs that were helpful to me.

#skip-changelog

### CODE REVIEW

**Comment by armcknight** [comment]
UI test `TopViewControllerTests.testPagesViewController()` failed in asan, reran it

**Comment by armcknight** [comment]
Two unit tests failing: 
- `SentryCrashInstallationReporterTests.testReportIsSentAndDeleted()`
- `-[SentryCrashReportStore_Tests testPruneReports]`
rerunning those

File: Samples/iOS-Swift/PerformanceBenchmarks/SentrySDKPerformanceBenchmarkTests.m [hunk 1]
```diff
@@ -10,7 +10,7 @@ @implementation SentrySDKPerformanceBenchmarkTests
 
 - (void)testCPUBenchmark
 {
-    XCTSkipIf(isSimulator() && !isDebugging());
+    //    XCTSkipIf(isSimulator() && !isDebugging());
```
Comment (brustolin) [inline Samples/iOS-Swift/PerformanceBenchmarks/SentrySDKPerformanceBenchmarkTests.m:13]: m: IMO commented code without an explanation has no value.
Maybe this was forgotten here by mistake.
We should either remove this or add an explanation.

File: Sources/Sentry/Profiling/SentryLaunchProfiling.m [hunk 2]
```diff
@@ -20,12 +20,9 @@
 #    import "SentryTransactionContext.h"
 
 BOOL isTracingAppLaunch;
-SentryId *_Nullable appLaunchTraceId;
-NSObject *appLaunchTraceLock;
-uint64_t appLaunchSystemTime;
 NSString *const kSentryLaunchProfileConfigKeyTracesSampleRate = @"traces";
 NSString *const kSentryLaunchProfileConfigKeyProfilesSampleRate = @"profiles";
-SentryTracer *_Nullable launchTracer;
+static SentryTracer *_Nullable launchTracer;
```
Comment (brustolin) [inline Sources/Sentry/Profiling/SentryLaunchProfiling.m:25]: l: I believe that without an `NS_ASSUME_NONNULL_BEGIN` in the file, everything is assumed nullable. I understand that it does not affect the code, but I prefer clear, less cluttered code as much as possible.

**Review by brustolin — APPROVED** [review]
Looks good.

File: Samples/iOS-Swift/PerformanceBenchmarks/SentrySDKPerformanceBenchmarkTests.m (hunk 1 above)
Comment (armcknight) [reply Samples/iOS-Swift/PerformanceBenchmarks/SentrySDKPerformanceBenchmarkTests.m:13]: Oops, yes this was a mistake

File: Sources/Sentry/Profiling/SentryLaunchProfiling.m (hunk 2 above)
Comment (armcknight) [reply Sources/Sentry/Profiling/SentryLaunchProfiling.m:25]: Actually I need to add those macros, so this _Nullable will need to stay as well

**Review by philipphofmann — APPROVED** [review]
LGTM

**Pull request merged**